homeassistant/components/zamg/coordinator.py

```diff
@@ -0,0 +1,46 @@
+"""Data Update coordinator for ZAMG weather data."""
+from __future__ import annotations
+
+from zamg import ZamgData as ZamgDevice
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import CONF_STATION_ID, DOMAIN, LOGGER, MIN_TIME_BETWEEN_UPDATES
+
+
+class ZamgDataUpdateCoordinator(DataUpdateCoordinator[ZamgDevice]):
+    """Class to manage fetching ZAMG weather data."""
+
+    config_entry: ConfigEntry
+    data: dict = {}
+
+    def __init__(self, hass: HomeAssistant, *, entry: ConfigEntry) -> None:
+        """Initialize global ZAMG data updater."""
+        self.zamg = ZamgDevice(session=async_get_clientsession(hass))
+        self.zamg.set_default_station(entry.data[CONF_STATION_ID])
+        super().__init__(
+            hass,
+            LOGGER,
+            name=DOMAIN,
+            update_interval=MIN_TIME_BETWEEN_UPDATES,
+        )
+
+    def update_listeners(self) -> None:
+        """Call update on all listeners."""
+        for update_callback in self._listeners:
+            update_callback()
```

Comment:
Where is this used? I couldn't find any reference.

Reply:
nowhere -> removed